openproblems/tools/normalize.py

```diff
@@ -52,3 +52,15 @@ def sqrt_cpm(adata: ad.AnnData) -> ad.AnnData:
     _cpm(adata)
     adata.X = scprep.transform.sqrt(adata.X)
     return adata
+
+
+@decorators.normalizer
+def preprocess_logCPM_1kHVG(adata: ad.AnnData) -> ad.AnnData:
+
+    log_cpm(adata)
+    sc.pp.highly_variable_genes(adata, n_top_genes=1000, flavor="cell_ranger")
+    adata = adata[:, adata.var["highly_variable"]]
+    sc.tl.pca(adata, n_comps=50, svd_solver="arpack")
+    adata.obsm["X_input"] = adata.obsm["X_pca"]
```

Comment:
I don't think this actually makes sense inside a normalizer -- PCA is a preprocessing step but not a normalization step (the normalizer should only touch `adata.X`). Can we move PCA back inside the methods rather than inside the normalizer?

Most obvious problem here is the use of `preprocess_logCPM_1kHVG(adata, obsm="mode2")` which would compute the PCA but not store it anywhere due to the way we handle normalization of `obsm` which is by creating a new AnnData, normalizing `.X`, and pasting the normalized `.X` back onto `obsm` of the original.

Reply:
I vaguely remember this coming from Daniel but yes, happy to move it.